src/arrayMethodSort.js

```diff
@@ -1,11 +1,27 @@
 'use strict';
 
-/**
- * Implement method Sort
- */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const ARRAY = [...this];
+    let cmpFunc = compareFunction;
+
+    if (!compareFunction) {
+      cmpFunc = (a, b) => String(a) > String(b);
+    }
+
+    for (let i = 0; i < ARRAY.length - 1; i++) {
+      for (let j = 0; j < ARRAY.length - 1 - i; j++) {
+        if (cmpFunc(ARRAY[j], ARRAY[j + 1]) > 0) {
+          [ARRAY[j], ARRAY[j + 1]] = [ARRAY[j + 1], ARRAY[j]];
+        }
+      }
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      this[i] = ARRAY[i];
+    }
+
+    return this;
```

Comment:
The method currently sorts a copy of the array and then copies the sorted elements back to the original array. This works, but it's not efficient. It would be better to sort the array in place without creating an additional copy. If you do need to create a copy, remember to return the copy instead of the original array to avoid side effects.